patches/esm+3.2.25.patch

```diff
@@ -1,9 +1,9 @@
?R.module.fsReadJSON:R.module.fsReadJSON=(function(){"use strict";return function(e){var t=Ri(e,"utf8");return null===t?null:fn(t)}})(),Cc=R.inited?R.module.fsReadJSON6:R.module.fsReadJSON6=(function(){"use strict";return function(e){var t=Ri(e,"utf8");return null===t?null:Ei(t)}})(),Oc=R.inited?R.module.fsReaddir:R.module.fsReaddir=(function(){"use strict";return function(e){if("string"==typeof e)try{return Jr(e)}catch(e){}return null}})(),Tc=s(0),Mc=46,Lc=en.OPTIONS,Dc=b.PACKAGE_RANGE,Fc=b.PACKAGE_VERSION,jc="*",Vc=ko.ERR_INVALID_ESM_OPTION,Gc=ko.ERR_UNKNOWN_ESM_OPTION,Bc=".esmrc",Uc="package.json",Wc=[".mjs",".cjs",".js",".json"],qc={await:!1,cache:!0,cjs:{cache:!1,dedefault:!1,esModule:!1,extensions:!1,mutableNamespace:!1,namedExports:!1,paths:!1,topLevelReturn:!1,vars:!1},debug:!1,force:!1,mainFields:["main"],mode:1,sourceMap:void 0,wasm:!1},zc={cjs:{cache:!0,dedefault:!1,esModule:!0,extensions:!0,mutableNamespace:!0,namedExports:!0,paths:!0,topLevelReturn:!1,vars:!0},mode:2};class Hc{constructor(e,t,r){r=Hc.createOptions(r);var i="";"string"==typeof r.cache?i=we(e,r.cache):!1!==r.cache&&(i=e+Ee+"node_modules"+Ee+".cache"+Ee+"esm");var n=R.package.dir;if(!n.has(i)){var s={buffer:null,compile:null,meta:null},a=null,o=new Map,u=null;if(""!==i){for(var l=Oc(i),c=!1,p=!1,h=!1,f=0,d=null==l?0:l.length;f<d;f++){var m=l[f];if(kc(m))o.set(m,null);else if(m.charCodeAt(0)===Mc)if(".data.blob"===m)c=!0;else if(".data.json"===m)h=!0;else if(".dirty"===m){p=!0;break}}var v=p,g=null;if(h&&!v&&(g=Nc(i+Ee+".data.json"),v=null===g||!M(g,"version")||g.version!==Fc||!M(g,"meta")||!D(g.meta)),v&&(c=!1,h=!1,o=new Map,p&&vn(i+Ee+".dirty"),(function(e){for(var t=we(e,"../@babel/register"),r=Oc(t),i=0,n=null==r?0:r.length;i<n;i++){var s=r[i];Ic(s)&&vn(t+Ee+s)}})(i)),c&&(a=Ri(i+Ee+".data.blob")),h){var y=g.meta,x=T(y);u=new Map;for(var b=0,w=null==x?0:x.length;b<w;b++){var E=x[b];u.set(E,y[E])}}}null===a&&(a=zr.alloc(0)),null===u&&(u=new Map),s.buffer=a,s.compile=o,s.meta=u,n.set(i,s)}this.cache=n.get(i),this.cachePath=i,this.dirPath=e,this.options=r,this.range=t}clone(){var e=this.options,t=e.cjs,r=Pn({__proto__:Hc.prototype},this),i=Pn({},e);return i.cjs=Pn({},t),r.options=i,r}static get(e,t){"."===e&&(e=H());var r=Pc.state.package,i=r.cache;""!==e||i.has("")||i.set("",new Hc("",Dc,{cache:!1,cjs:{topLevelReturn:!0}}));var n=(function e(t,r){var i=Pc.state.package,n=i.cache,s=i.default,a=null;if(n.has(t)&&(a=n.get(t),null!==a||void 0===r.forceOptions))return a;if("node_modules"===me(t))return n.set(t,null),null;if(a=s&&s.options.force?s.clone():(function(t,r){var i,n=t+Ee+Bc,s=Ac(n)?Ri(n,"utf8"):null,a=null!==s;a?s=Ei(s):n=$o(n,Hn,!1,Wc);var o=r.forceOptions;if(r.forceOptions=void 0,""!==n&&!a)if(a=!0,Ic(n))s=Cc(n);else{var u=Pc.state.package.cache,l=R.moduleState,c=l.parsing;i=new Hc(t,jc,{cache:Hc.createOptions(o).cache}),l.parsing=!1,u.set(t,i);try{i.options=Hc.createOptions(_c(n,null).module.exports)}finally{u.set(t,null),l.parsing=c}}var p,h=t+Ee+Uc,f=Ac(h)?Ri
```

Comment:
How did you make this patch? Is your work available in some sort of some source form?

Reply:
> How did you make this patch? Is your work available in some sort of some source form?

I pushed it to agoric-labs/esm#1